fedimint-api/src/db/mod.rs

```diff
@@ -212,46 +220,97 @@ impl DecodingError {
     }
 }
 
+// Given a database key prefix enum iterator, this function ensures that its prefix values
+// are unique within the enum scope.
+pub fn ensure_unique_key_prefixes<T: std::iter::Iterator + Clone>(db_key_prefix_iter: T)
+where
+    <T as Iterator>::Item: std::convert::Into<u8>,
+{
+    let prefixes: std::collections::BTreeSet<u8> = db_key_prefix_iter
+        .clone()
+        .map(|prefix| prefix.into())
+        .collect();
+
+    let result = prefixes.len();
+    let expected = db_key_prefix_iter.count();
+
+    assert_eq!(result, expected);
+}
+
```

Comment:
[This approach](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7e516f1e05e5183ab0eb30d1a23162b5) would automatically prevent duplicate prefixes.